src/ripple/rpc/handlers/Manifest.cpp

```diff
@@ -0,0 +1,60 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Dev Null Productions
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+namespace ripple {
+Json::Value doManifest (RPC::Context& context)
+{
+    auto& params = context.params;
+    Json::Value ret;
+
+    if(!params.isMember(jss::public_key))
+        return RPC::missing_field_error (jss::public_key);
+
+    auto pk = parseBase58<PublicKey>(TokenType::NodePublic, params[jss::public_key].asString());
+    if(!pk)
+    {
+        RPC::inject_error(rpcINVALID_PARAMS, ret);
+        return ret;
+    }
+
+    // first attempt to use params as ephemeral key,
+    // if this lookup succeeds master key will be returned,
+    // else pk will just be returned and we will assume that
+    // is master key anyways
+    auto mk = context.app.validatorManifests().getMasterKey(*pk);
+
+    auto ek = context.app.validatorManifests().getSigningKey(mk);
+
+    // if ephemeral key not found, we don't have specified manifest
+    if(ek == mk)
```

Comment:
This is a particularly confusing way of determining that we don't have a given manifest; not your fault, the API for this kind of thing is just awkward and it's something we ought to look at improving.

It also results in returning an empty JSON object and that _is_ something we can address here and now: we should, at the very least, return something helpful, e.g. the key that we searched for.

Consider if the returned JSON was:

```
{
    "requested": "nHDpbBXLzgktPjgcEEpb1Q9YgNDnZBPdHRXHza8PNLbUpGFxBZxV",
    /** These appear only if the manifest was found */
    "manifest" : "the actual manifest in hex",
    "details" : {
        "master_key" : "nHDpbBXLzgktPjgcEEpb1Q9YgNDnZBPdHRXHza8PNLbUpGFxBZxV",
        "ephemeral_key" : "n9LdwcMvLtsgpVcNeQJDCTQWbd46AY1i71pQPiJPsjnjBYuPWMu7",
        "sequence" : 1,
        "domain" : "area51.number8.us"
    }
}
```

Reply:
Updated the return JSON to match the suggested format and to always return the 'requested' key even in the case of the failed lookup